### PR TITLE
Update Keras application docs

### PR DESCRIPTION
**Pull request by Molkree**
- added lost cautions
- fixed references according to https://github.com/tensorflow/tensorflow/commit/71964116c54e3f1ad9686b9fb4987d526a15d8a7

### CODE REVIEW

File: tensorflow/python/keras/applications/resnet.py [hunk 1]
```diff
@@ -538,13 +538,16 @@ def decode_predictions(preds, top=5):
...
+  Reference:
+  - [Deep Residual Learning for Image Recognition](
+      https://arxiv.org/abs/1512.03385) (CVPR 2015)
 
   Optionally loads weights pre-trained on ImageNet.
   Note that the data format convention used by the model is
   the one specified in your Keras config at `~/.keras/keras.json`.
+  
+  Caution: Be sure to properly pre-process your inputs to the application.
+  Please see `applications.resnet.preprocess_input` for an example.
```
Comment (kyscg) [inline tensorflow/python/keras/applications/resnet.py:550]: Good work, maybe add a link here?

https://www.tensorflow.org/api_docs/python/tf/keras/applications/resnet/preprocess_input

**Review by kyscg — CHANGES_REQUESTED** [review]

File: tensorflow/python/keras/applications/resnet.py (hunk 1 above)
Comment (Molkree) [reply tensorflow/python/keras/applications/resnet.py:550]: If we are going to add the link here, it should be added to all other nets too, I think.
Additionally I noticed that densenet, nasnet are missing caution too (and reference is not as in https://github.com/tensorflow/tensorflow/commit/71964116c54e3f1ad9686b9fb4987d526a15d8a7).
I can convert this PR to update all other networks in the folder?

Comment (Molkree) [reply tensorflow/python/keras/applications/resnet.py:550]: Revisiting this a few days later I remembered that this caution will actually generate proper link on TensorFlow site (see [here](https://www.tensorflow.org/api_docs/python/tf/keras/applications/ResNet101V2) for an example), so I'll mark this as resolved. Still need to updated other nets with the same change though.

**Comment by gbaned** [comment]
@Molkree This PR is in draft, any update on this? Please. Thanks!

**Review by tanzhenyu — DISMISSED** [review]

**Comment by Molkree** [comment]
> @Molkree This PR is in draft, any update on this? Please. Thanks!

Hi, @gbaned!
Sorry for the abandoned PR, I updated the rest of the docs (added more missing cautions in DenseNet and NASNet and standardized all references according to https://github.com/tensorflow/tensorflow/commit/71964116c54e3f1ad9686b9fb4987d526a15d8a7).

While at it I can also remove `from __future__ import`s if you are willing to accept it in the same PR (and if it's actually okay to do but I presume so as TF no longer supports Python 2).

File: tensorflow/python/keras/applications/efficientnet.py [hunk 2]
```diff
@@ -16,9 +16,9 @@
 # pylint: disable=missing-docstring
 """EfficientNet models for Keras.
 
-Reference paper:
-  - [EfficientNet: Rethinking Model Scaling for Convolutional Neural Networks]
-    (https://arxiv.org/abs/1905.11946) (ICML 2019)
+  Reference:
```
Comment (tanzhenyu) [inline tensorflow/python/keras/applications/efficientnet.py:19]: this should have the same indent as "Reference Paper".

File: tensorflow/python/keras/applications/mobilenet.py [hunk 3]
```diff
@@ -56,9 +56,10 @@
 |  1.0 MobileNet-128  |    64.4 %    |        529        |     4.2     |
 ------------------------------------------------------------------------
 
-Reference paper:
-  - [MobileNets: Efficient Convolutional Neural Networks for
-     Mobile Vision Applications](https://arxiv.org/abs/1704.04861)
+Reference:
+  - [MobileNets: Efficient Convolutional Neural Networks
+     for Mobile Vision Applications](
+      https://arxiv.org/abs/1704.04861)
```
Comment (tanzhenyu) [inline tensorflow/python/keras/applications/mobilenet.py:62]: No need for this

File: tensorflow/python/keras/applications/efficientnet.py (hunk 2 above)
Comment (Molkree) [reply tensorflow/python/keras/applications/efficientnet.py]: fixed this one.

File: tensorflow/python/keras/applications/mobilenet.py (hunk 3 above)
Comment (Molkree) [reply tensorflow/python/keras/applications/mobilenet.py:62]: Do you mean splitting the link into 3 lines? I just copied it from line 98 below, but I can make it two lines again. (Not sure why it's like this below)

**Review by tanzhenyu — APPROVED** [review]

**Pull request merged**